wgengine/router_openbsd.go

```diff
@@ -168,12 +168,94 @@ func (r *openbsdRouter) Close() error {
 		r.logf("failed to restore system resolv.conf: %v", err)
 	}
 
-	// TODO(mbaillie): wipe routes
+	return nil
+}
+
+const (
+	tsConf     = "/etc/resolv.tailscale.conf"
+	backupConf = "/etc/resolv.pre-tailscale-backup.conf"
+	resolvConf = "/etc/resolv.conf"
+)
+
+func (r *openbsdRouter) replaceResolvConf(servers []net.IP, domains []string) error {
+	if len(servers) == 0 {
+		return r.restoreResolvConf()
+	}
+
+	// Write the tsConf file.
+	buf := new(bytes.Buffer)
+	fmt.Fprintf(buf, "# resolv.conf(5) file generated by tailscale\n")
+	fmt.Fprintf(buf, "# DO NOT EDIT THIS FILE BY HAND -- CHANGES WILL BE OVERWRITTEN\n\n")
+	for _, ns := range servers {
+		fmt.Fprintf(buf, "nameserver %s\n", ns)
+	}
+	if len(domains) > 0 {
+		fmt.Fprintf(buf, "search "+strings.Join(domains, " ")+"\n")
+	}
+	f, err := ioutil.TempFile(filepath.Dir(tsConf), filepath.Base(tsConf)+".*")
+	if err != nil {
+		return err
+	}
+	f.Close()
+	if err := atomicfile.WriteFile(f.Name(), buf.Bytes(), 0644); err != nil {
+		return err
+	}
+	os.Chmod(f.Name(), 0644) // ioutil.TempFile creates the file with 0600.
```

Comment:
You're only using TempFile to get a random name, though? atomicfile is doing 0644 and renaming the file into place. So the permissions should still be 0644. I don't think this line is doing anything.